### PR TITLE
graphql: support authenticated servers

### PR DESCRIPTION
**Pull request by tsiq-karold**

### CODE REVIEW

File: webtau-groovy/src/main/groovy/org/testingisdocumenting/webtau/cfg/GroovyConfigBasedHttpConfiguration.groovy [hunk 1]
```diff
@@ -49,7 +50,9 @@ class GroovyConfigBasedHttpConfiguration implements HttpConfiguration {
         }
 
         return HttpConfigurations.withDisabledConfigurations {
-            return headerProvider.get().call(fullUrl, passedUrl, given) as HttpHeader
+            HttpValidationHandlers.withDisabledHandlers {
```
Comment (MykolaGolubyev) [inline webtau-groovy/src/main/groovy/org/testingisdocumenting/webtau/cfg/GroovyConfigBasedHttpConfiguration.groovy:53]: Now I wonder if we are going to have the same issue in Java. Could you please add java full example in a separate PR when you have time?

File: webtau-http/src/main/java/org/testingisdocumenting/webtau/http/validation/HttpValidationHandlers.java [hunk 2]
```diff
@@ -26,6 +26,7 @@
 public class HttpValidationHandlers {
     private static final List<HttpValidationHandler> globalHandlers = ServiceLoaderUtils.load(HttpValidationHandler.class);
     private static final ThreadLocal<List<HttpValidationHandler>> localHandlers = ThreadLocal.withInitial(ArrayList::new);
+    private static final ThreadLocal<Boolean> enabled = ThreadLocal.withInitial(() -> true);
```
Comment (MykolaGolubyev) [inline webtau-http/src/main/java/org/testingisdocumenting/webtau/http/validation/HttpValidationHandlers.java:29]: missing copyright update

File: webtau-test-server/src/main/java/org/testingisdocumenting/webtau/http/testserver/GraphQLResponseHandler.java [hunk 3]
```diff
@@ -76,11 +111,19 @@ public void handle(String url, Request baseRequest, HttpServletRequest request,
         } else {
             response.setStatus(405);
         }
+    }
 
-        baseRequest.setHandled(true);
+    private boolean isAuthorised(HttpServletRequest request) {
```
Comment (MykolaGolubyev) [inline webtau-test-server/src/main/java/org/testingisdocumenting/webtau/http/testserver/GraphQLResponseHandler.java:116]: I know header is called "authorization", but I think what is happening here is Authentication?

Comment (tsiq-karold) [reply webtau-test-server/src/main/java/org/testingisdocumenting/webtau/http/testserver/GraphQLResponseHandler.java:116]: Yes, it's weird and confusing but it's not my invention: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization

File: webtau-groovy/src/main/groovy/org/testingisdocumenting/webtau/cfg/GroovyConfigBasedHttpConfiguration.groovy (hunk 1 above)
Comment (tsiq-karold) [reply webtau-groovy/src/main/groovy/org/testingisdocumenting/webtau/cfg/GroovyConfigBasedHttpConfiguration.groovy:53]: I created an equivalent Java test and we definitely have the same problem :(

Comment (tsiq-karold) [reply webtau-groovy/src/main/groovy/org/testingisdocumenting/webtau/cfg/GroovyConfigBasedHttpConfiguration.groovy:53]: I'll do that in a separate PR.

**Pull request merged**